test/nodes/test_link_content_fetcher.py

```diff
@@ -0,0 +1,370 @@
+    """
+    Ensures an InvalidURL exception is raised when URL is missing.
+    """
+    pre_processor_mock = Mock()
+    pre_processor_mock.process.return_value = [Document("Sample content from webpage")]
+
+    retriever_no_url = LinkContentFetcher(processor=pre_processor_mock)
+    with pytest.raises(requests.exceptions.InvalidURL, match="Invalid or missing URL"):
+        retriever_no_url.fetch(url="")
+
+
+@pytest.mark.unit
+def test_fetch_invalid_url(caplog, mocked_requests, mocked_article_extractor):
+    """
+    Ensures an InvalidURL exception is raised when the URL is invalid.
+    """
+    url = "this-is-invalid-url"
+
+    r = LinkContentFetcher()
+    with pytest.raises(requests.exceptions.InvalidURL):
+        r.fetch(url=url)
+
+
+@pytest.mark.unit
+def test_fetch_no_preprocessor(mocked_requests, mocked_article_extractor):
+    """
+    Checks if the LinkContentFetcher can fetch content without a preprocessor.
+    """
+    url = "https://www.example.com"
+    r = LinkContentFetcher()
+
+    result_no_preprocessor = r.fetch(url=url)
+
+    assert len(result_no_preprocessor) == 1
+    assert isinstance(result_no_preprocessor[0], Document)
+    assert result_no_preprocessor[0].content == "Sample content from webpage"
+
+
+@pytest.mark.unit
+def test_fetch_correct_arguments(mocked_requests, mocked_article_extractor):
+    """
+    Ensures that requests.get is called with correct arguments.
+    """
+    url = "https://www.example.com"
+
+    r = LinkContentFetcher()
+    r.fetch(url=url)
+
+    # Check the arguments that requests.get was called with
+    args, kwargs = mocked_requests.get.call_args
+    assert args[0] == url
+    assert kwargs["timeout"] == 3
+    assert kwargs["headers"] == r.REQUEST_HEADERS
+
+    # another variant
+    url = "https://deepset.ai"
+    r.fetch(url=url, timeout=10)
+    # Check the arguments that requests.get was called with
+    args, kwargs = mocked_requests.get.call_args
+    assert args[0] == url
+    assert kwargs["timeout"] == 10
+    assert kwargs["headers"] == r.REQUEST_HEADERS
+
+
+@pytest.mark.unit
+def test_fetch_default_empty_content(mocked_requests):
+    """
+    Checks handling of content extraction returning empty content.
+    """
+    url = "https://www.example.com"
+    timeout = 10
+    content_text = ""
+    r = LinkContentFetcher()
+
+    with patch("boilerpy3.extractors.ArticleExtractor.get_content", return_value=content_text):
+        result = r.fetch(url=url, timeout=timeout)
+
+    assert "text" not in result
+    assert isinstance(result, list) and len(result) == 1
+    assert isinstance(result[0], Document)
+    assert result[0].content == content_text
+
+
+@pytest.mark.unit
+def test_fetch_exception_during_content_extraction_no_raise_on_failure(caplog, mocked_requests):
+    """
+    Checks the behavior when there's an exception during content extraction, and raise_on_failure is set to False.
+    """
+    caplog.set_level(logging.DEBUG)
```

Comment:
```suggestion
    caplog.set_level(logging.WARNING)
```
Perhaps this and the following should be changed?